src/requests/typings/deliveryFunction.ts

```diff
@@ -50,13 +53,10 @@ export type DeliveryFunctionEventContext<P extends Record<string, any> = Record<
   appInstallationParameters: P
 }
 
-export type DeliveryFunctionEventHandlers = {
-  [DeliveryFunctionRequestEventType.GRAPHQL_FIELD_MAPPING]: (
-    event: GraphQLFieldTypeMappingRequest,
-    context: DeliveryFunctionEventContext
-  ) => Promise<GraphQLFieldTypeMappingResponse>
-  [DeliveryFunctionRequestEventType.GRAPHQL_QUERY]: (
-    event: GraphQLQueryRequest,
-    context: DeliveryFunctionEventContext
-  ) => Promise<GraphQLQueryResponse>
-}
+export type DeliveryFunctionEventRequest = GraphQLFieldTypeMappingRequest | GraphQLQueryRequest
+export type DeliveryFunctionEventResponse = GraphQLFieldTypeMappingResponse | GraphQLQueryResponse
+
+export type DeliveryFunctionEventHandler = (
```

Comment:
We could add some minimal convenience for customers like so:

```
type DeliveryFunctionEventHandlers = {
  'graphql.field.mapping': {
    event: GraphQLFieldTypeMappingRequest
    response: GraphQLFieldTypeMappingResponse
  }
  'graphql.query': {
    event: GraphQLQueryRequest
    response: GraphQLQueryResponse
  }
}

type EventHandler<P, E extends DeliveryFunctionEventRequest, R extends DeliveryFunctionEventResponse> = (
  event: E,
  context: DeliveryFunctionEventContext<P>
) => Promise<R> | R
export type Handler<
  K extends keyof DeliveryFunctionEventHandlers = keyof DeliveryFunctionEventHandlers,
  P = any
> = EventHandler<P, DeliveryFunctionEventHandlers[K]['event'], DeliveryFunctionEventHandlers[K]['response']>
```

Which would enable some basic type safety on their end:

```
type MyParams = {
  foo: string
  bar: boolean
}

const mappingHandler: Handler<'graphql.field.mapping', MyParams> = (event, context) => {
  return {}
}
const queryHandler: Handler<'graphql.query'> = (event, context) => {
  return {}
}
```

Reply:
Yes, I was thinking about a helper function but was unsure where to put it. I think for now this should be sufficient!